packages/jsii-rosetta/lib/languages/default.ts

```diff
@@ -153,6 +153,22 @@ export abstract class DefaultVisitor<C> implements AstHandler<C> {
       context.report(unsup, `Use of ${ts.SyntaxKind[unsup.kind]} in an object literal is not supported.`);
     }
 
+    const anyMembersFunctions = node.properties.some((p) =>
+      ts.isPropertyAssignment(p)
+        ? isExpressionOfFunctionType(context.typeChecker, p.initializer)
+        : ts.isShorthandPropertyAssignment(p)
+        ? isExpressionOfFunctionType(context.typeChecker, p.name)
+        : false,
+    );
+
+    const inferredType = context.inferredTypeOfExpression(node);
+    if ((inferredType && isJsiiProtocolType(context.typeChecker, inferredType)) || anyMembersFunctions) {
+      context.report(
+        node,
+        `You cannot use an object literal to make an instance of an interface. Define a class instead.`,
```

Comment:
This message is intended for example authors, who are authoring in TypeScript. At that point, we're not even talking about Python yet.

(By the way, the same limitations [sorta] holds for C#, and *technically* it would be possible to define and instantiate an anonymous class implementing an interface in Java, but Rosetta doesn't support it. There's also no point to supporting it in Java as C# and Python won't be able to do the same trick)